### ISSUE TITLE
Automatic Select Best LR

### ISSUE THREAD

**Issue by JonnoFTW** [surmenok/keras_lr_finder#16]
How do I pick the the best learning rate from the LRFinder object?

If there was a function to do this in the library, then this object could be used with `LearningRateScheduler` to automatically pick the learning rate again after `n` epochs using the lr finder method.

Code could look something like:

```python
import numpy as np
def get_derivatives(self, sma):
    assert sma >= 1
    derivatives = [0] * sma
    for i in range(sma, len(self.lrs)):
        derivatives.append((self.losses[i] - self.losses[i - sma]) / sma)
    return derivatives

def get_best_lr(self, sma, n_skip_beginning=10, n_skip_end=5):
    derivatives = self.get_derivatives(sma)
    best_der_idx = np.argmax(derivatives[n_skip_beginning:-n_skip_end])
    return self.lrs[n_skip_beginning:-n_skip_end][best_der_idx]      
```

Alternatively, you can just pick the pick the learning rate associated with the best loss, ie `new_lr = self.lrs[np.argmin(self.losses)]`

**Comment by JonnoFTW** [comment]
As an additional step to close the loop, you can train your network for a large number of epochs, continuously adjusting to the best learning rate with something like this (perhaps this example could be added to the docs if you accept my pull request for this feature):

```python
def find_lr_epoch_end(epoch, logs):
    if epoch % 30 == 0:
        lrf = LRFinder(model)
        lrf.find_generator(train_gen, 0.0001, 1, 1, num_samples // batch_size)
        new_lr = lrf.get_best_lr(4)
        print("New lr:", new_lr)
        K.set_value(model.optimizer.lr, new_lr)
lrcb = LambdaCallback(on_epoch_end=find_lr_epoch_end)
model.fit_generator(train_gen, batch_size, num_samples, epochs=500, callbacks=[lrcb])
```
Alternatively you could make something similar to the `ReduceLROnPlataeu` except it uses this method instead of simply decaying since it may turn out that you need a large learning rate to escape a local minima.
